CSharpMath.Tests/PreTypesetting/MathListBuilderTest.cs

```diff
@@ -970,7 +973,32 @@ public class MathListBuilderTest {
 
       Assert.Single(list);
       var latex = MathListBuilder.MathListToString(list);
-      Assert.Equal(@"\begin{array}{lr}x^2&\: x<0\\ x^3&\: x\geq 0\end{array}", latex);
+      Assert.Equal(RemoveWhiteSpacesFromString(latex), input);
+    }
+
+    [Fact]
+    public void TestListToString() {
+      TestLatexToListAndBack(@"\int_a^b");
+    }
+
+    [Fact]
+    public void TestListToString1() {
+      TestLatexToListAndBack(@"\int_wdf=\int_{\partial w}f");
+    }
+
+    [Fact]
+    public void TestMatrixListToString() {
+      TestLatexToListAndBack(@"\begin{ vmatrix}\sin(x) &\cos(x)\\-\cos(x) &\sin(x)\end{ vmatrix}= 1", @"\left| \begin{matrix}\sin (x)&\cos (x)\\ -\cos (x)&\sin (x)\end{matrix}\right| =1");
+    }
+
+    private static void TestLatexToListAndBack(string input, string expected = "") {
+      var list = MathLists.FromString(input);
+      var str = MathListBuilder.MathListToString(list);
+      Assert.Equal(string.IsNullOrWhiteSpace(expected) ? RemoveWhiteSpacesFromString(input): RemoveWhiteSpacesFromString(expected), RemoveWhiteSpacesFromString(str));
+    }
+    
+    private static string RemoveWhiteSpacesFromString(string str) {
+      return Regex.Replace(str, RemoveWhiteSpacesPattern, string.Empty, RegexOptions.Multiline);
     }

```

Comment:
Helper functions are usually placed at the top to let code readers know what helper functions are defined

Reply:
Unresolved